types/react-instantsearch-core/index.d.ts

```diff
@@ -61,7 +71,7 @@ export interface ConnectorDescription<TProvided, TExposed> {
    * searchForFacetValuesResults holds the search for facet values results.
    */
   getProvidedProps(
-    this: React.Component<TExposed>,
+    this: React.Component<TExposed> & ConnectedWidget,
```

Comment:
You could probably add `<TProvided>` here and use that in the `ConnectedWidget` definition

Reply:
I did that first, but it caused a conflict in the `this` and then returned only `React.Component<TExposed> & ConnectedWidget`, and couldn't infer the returned props anymore. Maybe another time :)